lib/libsacd/dstdec/dst_init.c

```diff
@@ -80,8 +80,8 @@ RT:  Richard Theelen, PDSL-labs Eindhoven <r.h.m.theelen@philips.com>
 static void *MemoryAllocate(int NrOfElements, int SizeOfElement) 
 {
   void *Array;
-
-  if ((Array = _mm_malloc(NrOfElements * SizeOfElement, 16)) == NULL) 
+  posix_memalign(&Array, 16, NrOfElements * SizeOfElement);
+  if (Array == NULL)
```

Comment:
Is checking the pointer the correct thing for `posix_memalign`?
```suggestion
  if (posix_memalign(&Array, 16, NrOfElements * SizeOfElement) == 0)
```

`_mm_free` a few lines below has to be replaced with `free`.